sftp-client.c

```diff
@@ -1598,6 +1635,14 @@ do_upload(struct sftp_conn *conn, const char *local_path,
 	if (!preserve_flag)
 		a.flags &= ~SSH2_FILEXFER_ATTR_ACMODTIME;
 
+	/* *** ICL Modification ***  Encrypt local file before uploading. */
+	local_fd = write_gpg_encrypted_file(local_path, 1, NULL);
+	if (local_fd < 0) {
+		error("Unable to encrypt local file \"%s\" before uploading.", local_path);
+		return(-1);
+	}
+	/* *** */
+
```

Comment:
Is local_path edited to add the .iron suffix inside write_gpg_encrypted_file?  I must have seen that.